LabelBench/trainer/trainer_impl/uda.py

```diff
@@ -0,0 +1,63 @@
+import torch
+import torch.nn.functional as F
+from LabelBench.trainer.trainer_impl.pytorch_semi_trainer import PyTorchSemiTrainer
+import numpy as np
+from timm.loss import SoftTargetCrossEntropy
+
+
+class UDATrainer(PyTorchSemiTrainer):
+    trainer_name = "uda"
+    # Adapted from https://github.com/microsoft/Semi-supervised-learning/tree/main/semilearn/algorithms/uda.
+
+    def __init__(self, trainer_config, dataset, model_fn, model_config, metric, get_feature_fn):
+        """See `LabelBench.skeleton.trainer_skeleton.Trainer` for detailed documentation of the above arguments."""
+        super().__init__(trainer_config, dataset, model_fn, model_config, metric, get_feature_fn)
+        self.use_strong = True
+        self.unlabeled_idxs = None
+        self.consistency_loss = SoftTargetCrossEntropy()
+
+    def initialize_trainer(self):
+        self.unlabeled_idxs = torch.from_numpy(self.dataset.unlabeled_idxs()).long()
+
+    def train_step(self, model, img_l, target_l, class_weights, loss_fn, idx_u, img_uw, img_us, iter):
+        # TODO: documentation
```

Comment:
I'm not sure if we need the documentation here. Perhaps you could just take the documentation from the microsoft repo or remove the TODO?